libafl/src/corpus/mod.rs

```diff
@@ -148,6 +148,18 @@ pub trait Corpus: UsesInput + Serialize + for<'de> Deserialize<'de> {
     }
 }
 
+/// Trait for types which track the current corpus index
+pub trait HasCorpusStatus {
```

Comment:
Is `Has` really all it does here(?) maybe `trait CurrentCorpusIdx` or `TracksCorpusIdx` or something would be more meaningful?